pymc_marketing/mmm/models/components/base.py

```diff
@@ -0,0 +1,155 @@
+        if not new_priors:
+            available_priors = list(self.variable_mapping.values())
+            warnings.warn(
+                f"No priors were updated. Available parameters are {available_priors}",
+                UserWarning,
+                stacklevel=2,
+            )
+
+        self.function_priors.update(new_priors)
+
+    @property
+    def model_config(self) -> dict[str, Any]:
+        return {
+            variable_name: self.function_priors[parameter_name]
+            for parameter_name, variable_name in self.variable_mapping.items()
+        }
+
+    def _checks(self) -> None:
+        self._has_all_attributes()
+        self._function_works_on_instances()
+        self._has_defaults_for_all_arguments()
+
+    def _has_all_attributes(self) -> None:
+        if not hasattr(self, "prefix"):
+            raise NotImplementedError("prefix must be implemented in the subclass")
+
+        if not hasattr(self, "default_priors"):
+            raise NotImplementedError(
+                "default_priors must be implemented in the subclass"
+            )
+
+        if not hasattr(self, "function"):
+            raise NotImplementedError("function must be implemented in the subclass")
+
+    def _has_defaults_for_all_arguments(self) -> None:
+        function_signature = signature(self.function)
+
+        # Remove the first one
+        parameters_that_need_priors = set(
+            list(function_signature.parameters.keys())[1:]
+        )
+        parameters_with_priors = set(self.default_priors.keys())
+
+        missing_priors = parameters_that_need_priors - parameters_with_priors
+        missing_parameters = parameters_with_priors - parameters_that_need_priors
+
+        if missing_priors or missing_parameters:
+            raise ParameterPriorException(missing_priors, missing_parameters)
+
+    def _function_works_on_instances(self) -> None:
+        n_parameters = len(signature(self.function).parameters)
+
+        n_default_parameters = len(self.default_priors)
+
+        if n_parameters - 1 == n_default_parameters:
+            return
+
+        self.function = self.__class__.function
+
+    @property
+    def variable_mapping(self) -> dict[str, str]:
+        return {
+            parameter: f"{self.prefix}_{parameter}"
+            for parameter in self.default_priors.keys()
+        }
+
+    def _create_distributions(self, dim_name: str) -> dict[str, pt.TensorVariable]:
+        distributions: dict[str, pt.TensorVariable] = {}
+        for parameter_name, variable_name in self.variable_mapping.items():
+            parameter_prior = self.function_priors[parameter_name]
+
+            distribution = _get_distribution(
+                dist=parameter_prior,
+            )
+
+            distributions[parameter_name] = distribution(
+                name=variable_name,
+                dims=dim_name,
+                **parameter_prior["kwargs"],
+            )
+
+        return distributions
+
+    def apply(self, x):
```

Comment:
Good point but we already talk about this and probably in the future after merge this PR, play with the `dims` will be very easy. 

Meaning, better not to attach directly to the `channel`, because probably dims will change, allowing create hierarchies or others!

Reply:
I am leaning toward making it an argument with a default of "channel". 
I have an example above to show why